commons-statistics-descriptive/src/main/java/org/apache/commons/statistics/descriptive/Variance.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.statistics.descriptive;
+
+/**
+ * Computes the variance of a set of values. By default, the
+ * "sample variance" is computed. The definitional formula for sample
+ * variance is:
+ * <p>
+ * sum((x_i - mean)^2) / (n - 1)
+ * <p>This formula does not have good numerical properties, so this
+ * implementation does not use it to compute the statistic.
+ * <ul>
+ * <li> The {@link #accept(double)} method computes the variance using
+ * updating formulae based on West's algorithm, as described in
+ * <a href="http://doi.acm.org/10.1145/359146.359152"> Chan, T. F. and
+ * J. G. Lewis 1979, <i>Communications of the ACM</i>,
+ * vol. 22 no. 9, pp. 526-531.</a></li>
+ *
+ * <li> The {@link #of(double...)} method leverages the fact that it has the
+ * full array of values in memory to execute a two-pass algorithm.
+ * Specifically, this method uses the "corrected two-pass algorithm" from
+ * Chan, Golub, Levesque, <i>Algorithms for Computing the Sample Variance</i>,
+ * American Statistician, vol. 37, no. 3 (1983) pp. 242-247.</li></ul>
+ *
+ * Note that adding values using {@code accept} and then executing {@code getAsDouble} will
+ * sometimes give a different, less accurate, result than executing
```

Comment:
Are we able to say anything more about accuracy? Maybe the papers linked above have something we could add here?

At $work we are using @tdunning's tdigest (a Python port based on it) for one-pass algorithms, where researchers are validating the computation and accuracy of statistics. They used both the paper and also notes from the javadocs that talk about the performance/accuracy tradeoff in the algorithm. e.g. https://github.com/tdunning/t-digest/blob/72acae1e14962795888efc941516c2b4503a86af/core/src/main/java/com/tdunning/math/stats/TDigest.java#L34 & https://github.com/tdunning/t-digest/blob/72acae1e14962795888efc941516c2b4503a86af/core/src/main/java/com/tdunning/math/stats/ScaleFunction.java#L21

Maybe we don't need to be very specific here, and we can just expand  a bit more and quote the article authors?

Reply:
BTW, this can be a follow-up issue, no need to block this PR for that, nor the other tasks needed for GSoC, I think.

Reply:
I'll go through the papers once again and see if I can add some more information regarding accuracy. 👍